optimum/exporters/onnx/utils.py

```diff
@@ -165,14 +165,23 @@ def get_stable_diffusion_models_for_export(
     unet_onnx_config = onnx_config_constructor(pipeline.unet.config)
     models_for_export["unet"] = (pipeline.unet, unet_onnx_config)
 
-    # VAE
-    vae = copy.deepcopy(pipeline.vae)
-    vae.forward = lambda latent_sample: vae.decode(z=latent_sample)
+    # VAE Encoder
+    vae_encoder = copy.deepcopy(pipeline.vae)
+    vae_encoder.forward = lambda sample: {"latent_sample": vae_encoder.encode(x=sample)["latent_dist"].sample()}
```

Comment:
I have a hard time reading this but again am not familiar. Could you add a comment as to why it is necessary to override the forward with this?

Reply:
Yes the VAE should be exported in two different components (encoder and decoder) as they are used independently in the stable diffusion pipeline

Reply:
all good

Reply:
So the `vae_encoder.forward = lambda sample: {"latent_sample": vae_encoder.encode(x=sample)["latent_dist"].sample()}` corresponds to the two lines I linked? Could you add a link as a comment, e.g. https://github.com/huggingface/diffusers/blob/7c82a16fc14840429566aec40eb9e65aa57005fd/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py#L498 , if that is what the encoder represents?